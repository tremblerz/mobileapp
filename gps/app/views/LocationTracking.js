import BackgroundGeolocation from '@mauron85/react-native-background-geolocation';
import React, { Component } from 'react';
import {
  AppState,
  BackHandler,
  Dimensions,
  Image,
  ImageBackground,
  Linking,
  NativeModules,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import {
  PERMISSIONS,
  RESULTS,
  check,
  openSettings,
} from 'react-native-permissions';
import Pulse from 'react-native-pulse';
import { SvgXml } from 'react-native-svg';

import BackgroundImageAtRisk from '../../../shared/assets/images/backgroundAtRisk.png';
import exportImage from '../../../shared/assets/images/export.png';
import foreArrow from '../../../shared/assets/images/foreArrow.png';
import BackgroundImage from '../../../shared/assets/images/launchScreenBackground.png';
import settingsIcon from '../../../shared/assets/svgs/settingsIcon';
import StateAtRisk from '../../../shared/assets/svgs/stateAtRisk';
import StateNoContact from '../../../shared/assets/svgs/stateNoContact';
import StateUnknown from '../../../shared/assets/svgs/stateUnknown';
import { isPlatformAndroid, isPlatformiOS } from './../Util';
import { Button } from '../components/Button';
import { Typography } from '../components/Typography';
import Colors from '../constants/colors';
import fontFamily from '../constants/fonts';
import {
  CROSSED_PATHS,
  DEBUG_MODE,
  LOCATION_DATA,
  PARTICIPATE,
} from '../constants/storage';
import { Theme } from '../constants/themes';
import {
  GetStoreData,
  RemoveStoreData,
  SetStoreData,
} from '../helpers/General';
import { checkIntersect } from '../helpers/Intersect';
import languages from '../locales/languages';
import BackgroundTaskServices from '../services/BackgroundTaskService';
import LocationServices from '../services/LocationService';

const MAYO_COVID_URL = 'https://www.mayoclinic.org/coronavirus-covid-19';

const StateEnum = {
  UNKNOWN: 0,
  AT_RISK: 1,
  NO_CONTACT: 2,
  SETTING_OFF: 3,
};

const StateIcon = ({ status, size }) => {
  let icon;
  switch (status) {
    case StateEnum.UNKNOWN:
      icon = StateUnknown;
      break;
    case StateEnum.AT_RISK:
      icon = StateAtRisk;
      break;
    case StateEnum.NO_CONTACT:
      icon = StateNoContact;
      break;
    case StateEnum.SETTING_OFF:
      icon = StateUnknown;
      break;
  }
  return (
    <SvgXml xml={icon} width={size ? size : 80} height={size ? size : 80} />
  );
};

const height = Dimensions.get('window').height;

class LocationTracking extends Component {
  constructor(props) {
    super(props);

    if (isPlatformAndroid()) {
      StatusBar.setBackgroundColor(Colors.TRANSPARENT);
      StatusBar.setBarStyle('light-content');
      StatusBar.setTranslucent(true);
    }

    this.state = {
      appState: AppState.currentState,
      timer_intersect: null,
      isLogging: '',
      currentState: StateEnum.NO_CONTACT,
    };
    try {
      this.checkCurrentState();
    } catch (e) {
      // statements
      console.log(e);
    }
  }

  /*  Check current state
        1) determine if user has correct location permissions
        2) check if they are at risk -> checkIfUserAtRisk()
        3) set state accordingly */
  checkCurrentState() {
    // NEED TO TEST ON ANDROID
    let locationPermission;
    if (isPlatformiOS()) {
      locationPermission = PERMISSIONS.IOS.LOCATION_ALWAYS;
    } else {
      locationPermission = PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION;
    }

    // If user has location enabled & permissions, start logging
    GetStoreData(PARTICIPATE, false).then(isParticipating => {
      if (isParticipating) {
        check(locationPermission)
          .then(result => {
            switch (result) {
              case RESULTS.GRANTED:
                LocationServices.start();
                this.checkIfUserAtRisk();
                return;
              case RESULTS.UNAVAILABLE:
              case RESULTS.BLOCKED:
                console.log('NO LOCATION');
                LocationServices.stop();
                this.setState({ currentState: StateEnum.UNKNOWN });
            }
          })
          .catch(error => {
            console.log('error checking location: ' + error);
          });
      } else {
        this.setState({ currentState: StateEnum.SETTING_OFF });
        LocationServices.stop();
      }
    });
  }

  checkIfUserAtRisk() {
    BackgroundTaskServices.start();
    // If the user has location tracking disabled, set enum to match
    GetStoreData(PARTICIPATE, false).then(isParticipating => {
      if (isParticipating === false) {
        this.setState({
          currentState: StateEnum.SETTING_OFF,
        });
      }
      //Location enable
      else {
        this.crossPathCheck();
      }
    });
  }
  //Due to Issue 646 moved below code from checkIfUserAtRisk function
  crossPathCheck() {
    GetStoreData(DEBUG_MODE).then(dbgMode => {
      if (dbgMode != 'true') {
        // already set on 12h timer, but run when this screen opens too
        checkIntersect();
      }
      GetStoreData(CROSSED_PATHS).then(dayBin => {
        dayBin = JSON.parse(dayBin);
        if (dayBin !== null && dayBin.reduce((a, b) => a + b, 0) > 0) {
          console.log('Found crossed paths');
          this.setState({ currentState: StateEnum.AT_RISK });
        } else {
          console.log("Can't find crossed paths");
          this.setState({ currentState: StateEnum.NO_CONTACT });
        }
      });
    });
  }

  componentDidMount() {
    AppState.addEventListener('change', this.handleAppStateChange);
    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
    // refresh state if settings have changed
    this.unsubscribe = this.props.navigation.addListener('focus', () => {
      this.checkIfUserAtRisk();
    });
    GetStoreData(PARTICIPATE, false)
      .then(isParticipating => {
        if (isParticipating === 'true') {
          this.setState({
            isLogging: true,
          });
          this.willParticipate();
        } else {
          this.setState({
            isLogging: false,
            currentState: StateEnum.SETTING_OFF,
          });
        }
      })
      .catch(error => console.log(error));

    GetStoreData(LOCATION_DATA, false).then(locations => {
      if (Array.isArray(locations) && locations.length > 0) {
        NativeModules.SecureStorageManager.migrateExistingLocations(
          locations,
        ).then(() => {
          RemoveStoreData(LOCATION_DATA);
        });
      }
    });
  }

  componentWillUnmount() {
    AppState.removeEventListener('change', this.handleAppStateChange);
    clearInterval(this.state.timer_intersect);
    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
    this.unsubscribe();
  }

  // need to check state again if new foreground event
  // e.g. if user changed location permission
  handleAppStateChange = nextAppState => {
    if (
      this.state.appState.match(/inactive|background/) &&
      nextAppState === 'active'
    ) {
      this.checkCurrentState();
    }
    this.setState({ appState: nextAppState });
  };

  handleBackPress = () => {
    BackHandler.exitApp(); // works best when the goBack is async
    return true;
  };

  export() {
    this.props.navigation.navigate('ExportScreen', {});
  }

  import() {
    this.props.navigation.navigate('ImportScreen', {});
  }

  willParticipate = () => {
    SetStoreData(PARTICIPATE, 'true').then(() => {
      // Turn of bluetooth for v1
      //BroadcastingServices.start();
    });
    // Check and see if they actually authorized in the system dialog.
    // If not, stop services and set the state to !isLogging
    // Fixes tripleblindmarket/private-kit#129
    BackgroundGeolocation.checkStatus(({ authorization }) => {
      if (authorization === BackgroundGeolocation.AUTHORIZED) {
        LocationServices.start();
        this.setState({
          isLogging: true,
        });
      } else if (authorization === BackgroundGeolocation.NOT_AUTHORIZED) {
        LocationServices.stop();
        // Turn off bluetooth for v1
        //BroadcastingServices.stop(this.props.navigation);
        BackgroundTaskServices.stop();
        this.setState({
          isLogging: false,
        });
      }
    });
  };

  getBackground() {
    if (this.state.currentState === StateEnum.AT_RISK) {
      return BackgroundImageAtRisk;
    }
    return BackgroundImage;
  }

  settings() {
    this.props.navigation.navigate('SettingsScreen', {});
  }

  getSettingsBtn() {
    return (
      <TouchableOpacity
        style={styles.settingsContainer}
        onPress={() => {
          this.props.navigation.navigate('SettingsScreen');
        }}>
        {/* Is there is a reason there's this imageless image tag here? Can we delete it? */}
        <Image resizeMode={'contain'} />
        <SvgXml xml={settingsIcon} width={30} height={30} color='white' />
      </TouchableOpacity>
    );
  }

  getPulseIfNeeded() {
    if (this.state.currentState == StateEnum.NO_CONTACT) {
      return (
        <View style={styles.pulseContainer}>
          <Pulse
            image={{ exportImage }}
            color={Colors.PULSE_WHITE}
            numPulses={3}
            diameter={400}
            speed={20}
            duration={2000}
          />
          <StateIcon size={height} status={this.state.currentState} />
        </View>
      );
    }
    return (
      <View style={styles.pulseContainer}>
        <Text>Testdsfafasfsdafasfsadf</Text>
        <StateIcon size={height} status={this.state.currentState} />
      </View>
    );
  }

  getMainText() {
    switch (this.state.currentState) {
      case StateEnum.NO_CONTACT:
        return (
          <Typography style={styles.mainTextBelow}>
            {languages.t('label.home_no_contact_header')}
          </Typography>
        );
      case StateEnum.AT_RISK:
        return (
          <Typography style={styles.mainTextAbove}>
            {languages.t('label.home_at_risk_header')}
          </Typography>
        );
      case StateEnum.UNKNOWN:
        return (
          <Typography style={styles.mainTextBelow}>
            {languages.t('label.home_unknown_header')}
          </Typography>
        );
      case StateEnum.SETTING_OFF:
        return (
          <Text style={styles.mainTextBelow}>
            {languages.t('label.home_setting_off_header')}
          </Text>
        );
    }
  }

  getSubText() {
    switch (this.state.currentState) {
      case StateEnum.NO_CONTACT:
        return languages.t('label.home_no_contact_subtext');
      case StateEnum.AT_RISK:
        return languages.t('label.home_at_risk_subtext');
      case StateEnum.UNKNOWN:
        return languages.t('label.home_unknown_subtext');
      case StateEnum.SETTING_OFF:
        return languages.t('label.home_setting_off_subtext');
    }
  }
  getSubSubText() {
    switch (this.state.currentState) {
      case StateEnum.NO_CONTACT:
        return null;
      case StateEnum.AT_RISK:
        return languages.t('label.home_at_risk_subsubtext');
      case StateEnum.UNKNOWN:
        return null;
      case StateEnum.SETTING_OFF:
        return null;
    }
  }

  getCTAIfNeeded() {
    let buttonLabel;
    let buttonFunction;
    if (this.state.currentState === StateEnum.NO_CONTACT) {
      return;
    } else if (this.state.currentState === StateEnum.AT_RISK) {
      buttonLabel = languages.t('label.see_exposure_history');
      buttonFunction = () => {
        this.props.navigation.navigate('ExposureHistoryScreen');
      };
    } else if (this.state.currentState === StateEnum.UNKNOWN) {
      buttonLabel = languages.t('label.home_enable_location');
      buttonFunction = () => {
        openSettings();
      };
    } else if (this.state.currentState === StateEnum.SETTING_OFF) {
      buttonLabel = languages.t('label.home_enable_location');
      buttonFunction = () => {
        this.settings();
      };
    }
    return (
      <Button
        label={buttonLabel}
        onPress={() => buttonFunction()}
        style={styles.buttonContainer}
      />
    );
  }

  getMayoInfoPressed() {
    Linking.openURL(MAYO_COVID_URL);
  }

  render() {
    const hasPossibleExposure = this.state.currentState === StateEnum.AT_RISK;
    return (
      <Theme use={hasPossibleExposure ? 'charcoal' : 'violet'}>
        <ImageBackground
          source={this.getBackground()}
          style={styles.backgroundImage}>
          <StatusBar
            barStyle='light-content'
            backgroundColor='transparent'
            translucent
          />
          {this.getPulseIfNeeded()}

          <View style={styles.mainContainer}>
            <View style={styles.contentAbovePulse}>
              {hasPossibleExposure && this.getMainText()}
              <Typography style={styles.subsubheaderText}>
                {this.getSubSubText()}
              </Typography>
            </View>
            <View style={styles.contentBelowPulse}>
              {!hasPossibleExposure && this.getMainText()}
              <Typography style={styles.subheaderText}>
                {this.getSubText()}
              </Typography>
              {this.getCTAIfNeeded()}
            </View>
          </View>

          <View>
            <TouchableOpacity
              onPress={this.getMayoInfoPressed.bind(this)}
              style={styles.mayoInfoRow}>
              <View style={styles.mayoInfoContainer}>
                <Typography
                  style={styles.mainMayoHeader}
                  onPress={() => Linking.openURL(MAYO_COVID_URL)}>
                  {languages.t('label.home_mayo_link_heading')}
                </Typography>
                <Typography
                  style={styles.mainMayoSubtext}
                  onPress={() => Linking.openURL(MAYO_COVID_URL)}>
                  {languages.t('label.home_mayo_link_label')}
                </Typography>
              </View>
              <View style={styles.arrowContainer}>
                <Image source={foreArrow} style={this.arrow} />
              </View>
            </TouchableOpacity>
          </View>
          {this.getSettingsBtn()}
        </ImageBackground>
      </Theme>
    );
  }
}

const PULSE_GAP = 80;

const styles = StyleSheet.create({
  backgroundImage: {
    width: '100%',
    height: '100%',
    resizeMode: 'cover',
    flex: 1,
    justifyContent: 'flex-end',
  },
  mainContainer: {
    position: 'absolute',
    // resizeMode: 'contain',
    // aligns the center of the main container with center of pulse
    // so that two `flex: 1` views will be have a reasonable chance at natural
    // flex flow for above and below the pulse.
    top: '-10%',
    left: 0,
    right: 0,
    height: '100%',
    paddingHorizontal: '12%',
    paddingBottom: 12,
  },
  contentAbovePulse: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingBottom: PULSE_GAP / 2,
  },
  contentBelowPulse: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingTop: PULSE_GAP,
  },
  settingsContainer: {
    position: 'absolute',
    top: 0,
    marginTop: '14%',
    marginRight: '7%',
    alignSelf: 'flex-end',
  },
  buttonContainer: {
    marginTop: 24,
    height: 54, // fixes overlaying buttons on really small screens
  },
  pulseContainer: {
    position: 'absolute',
    resizeMode: 'contain',
    height: '100%',
    top: '-13%',
    left: 0,
    right: 0,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  mainTextAbove: {
    textAlign: 'center',
    lineHeight: 34,
    marginBottom: 24,
    color: Colors.WHITE,
    fontSize: 28,
    fontFamily: fontFamily.primaryMedium,
  },
  mainTextBelow: {
    textAlign: 'center',
    lineHeight: 34,
    color: Colors.WHITE,
    fontSize: 26,
    fontFamily: fontFamily.primaryMedium,
    marginBottom: 24,
  },
  subheaderText: {
    marginBottom: 24,
    textAlign: 'center',
    lineHeight: 24.5,
    color: Colors.WHITE,
    fontSize: 18,
    fontFamily: fontFamily.primaryRegular,
  },
  subsubheaderText: {
    textAlign: 'center',
    lineHeight: 24.5,
    color: Colors.WHITE,
    fontSize: 16,
    fontFamily: fontFamily.primaryLight,
    marginBottom: 24,
  },
  mayoInfoRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  mayoInfoContainer: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignContent: 'flex-end',
    padding: 20,
  },
  mainMayoHeader: {
    textAlign: 'left',
    color: Colors.MISCHKA,
    fontSize: 18,
    fontFamily: fontFamily.primaryBold,
  },
  mainMayoSubtext: {
    textAlign: 'left',
    color: Colors.MISCHKA,
    fontSize: 18,
    fontFamily: fontFamily.primaryRegular,
  },
  arrowContainer: {
    alignSelf: 'center',
    paddingRight: 20,
    paddingLeft: 20,
  },
});

export default LocationTracking;
