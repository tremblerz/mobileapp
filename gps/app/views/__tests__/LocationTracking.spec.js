import 'react-native';

import BackgroundGeolocation from '@mauron85/react-native-background-geolocation';
import { act, render } from '@testing-library/react-native';
import React from 'react';

import LocationTracking from '../LocationTracking';

const navigationMock = {
  addListener: jest.fn(),
};

beforeEach(() => {
  BackgroundGeolocation.checkStatus.mockReturnValue(jest.fn());
  BackgroundGeolocation.configure.mockReturnValue(jest.fn());
});

it('renders correctly', async () => {
  const { asJSON } = render(<LocationTracking navigation={navigationMock} />);

  await act(async () => {
    expect(asJSON()).toMatchSnapshot();
  });
});
