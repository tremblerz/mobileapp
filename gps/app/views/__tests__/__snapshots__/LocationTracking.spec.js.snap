// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "justifyContent": "flex-end",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../shared/assets/images/launchScreenBackground.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "resizeMode": "contain",
          "right": 0,
          "top": "-13%",
        }
      }
    >
      <SvgXml
        height={1334}
        width={1334}
        xml="<svg width=\\"850\\" height=\\"850\\" viewBox=\\"0 0 850 850\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<path d=\\"M425 465C447.091 465 465 447.091 465 425C465 402.909 447.091 385 425 385C402.909 385 385 402.909 385 425C385 447.091 402.909 465 425 465Z\\" fill=\\"white\\"/>
<path d=\\"M425 457C442.673 457 457 442.673 457 425C457 407.327 442.673 393 425 393C407.327 393 393 407.327 393 425C393 442.673 407.327 457 425 457Z\\" fill=\\"#BEBEBE\\"/>
<path d=\\"M426.373 429.756V427.292C429.005 426.816 431.581 424.968 431.581 421.608C431.581 418.304 429.145 416.12 425.421 416.12C421.725 416.12 419.765 418.276 419.065 420.628L421.725 421.692C422.173 420.012 423.377 418.78 425.309 418.78C427.129 418.78 428.333 419.676 428.333 421.524V421.916C428.333 424.268 426.457 425.304 423.433 425.416V429.756H426.373ZM424.973 436.28C426.681 436.28 427.325 435.468 427.325 434.404V433.732C427.325 432.668 426.681 431.856 424.973 431.856C423.265 431.856 422.621 432.668 422.621 433.732V434.404C422.621 435.468 423.265 436.28 424.973 436.28Z\\" fill=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"85.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"149.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"217.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"283.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"352.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"424.5\\" stroke=\\"white\\"/>
</svg>"
      />
    </View>
    <View
      style={
        Object {
          "height": "100%",
          "left": 0,
          "paddingBottom": 12,
          "paddingHorizontal": "12%",
          "position": "absolute",
          "right": 0,
          "top": "-10%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-end",
            "paddingBottom": 40,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans-Light",
              "fontSize": 16,
              "fontWeight": "normal",
              "lineHeight": 24.5,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-start",
            "paddingTop": 80,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans-Medium",
              "fontSize": 26,
              "fontWeight": "normal",
              "lineHeight": 34,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        >
          No known contact
        </Text>
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans",
              "fontSize": 18,
              "fontWeight": "normal",
              "lineHeight": 24.5,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        >
          Based on available data you havenâ€™t been near anyone reported positive for COVID-19.
        </Text>
      </View>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "flex-end",
              "flexDirection": "column",
              "justifyContent": "space-between",
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#E5E4E6",
                "fontFamily": "IBMPlexSans-Bold",
                "fontSize": 18,
                "fontWeight": "normal",
                "lineHeight": 24,
                "textAlign": "left",
                "writingDirection": "ltr",
              }
            }
            use="body1"
          >
            More COVID-19 information
          </Text>
          <Text
            style={
              Object {
                "color": "#E5E4E6",
                "fontFamily": "IBMPlexSans",
                "fontSize": 18,
                "fontWeight": "normal",
                "lineHeight": 24,
                "textAlign": "left",
                "writingDirection": "ltr",
              }
            }
            use="body1"
          >
            from the Mayo Clinic
          </Text>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../shared/assets/images/foreArrow.png",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      style={
        Object {
          "alignSelf": "flex-end",
          "marginRight": "7%",
          "marginTop": "14%",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
        }
      }
    >
      <Image
        resizeMode="contain"
      />
      <SvgXml
        color="white"
        height={30}
        width={30}
        xml="
<svg width=\\"24\\" height=\\"22\\" viewBox=\\"0 0 24 22\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <path d=\\"M23.625 17H9V15.125C9 14.5156 8.48438 14 7.875 14H5.625C4.96875 14 4.5 14.5156 4.5 15.125V17H0.375C0.140625 17 0 17.1875 0 17.375V18.125C0 18.3594 0.140625 18.5 0.375 18.5H4.5V20.375C4.5 21.0312 4.96875 21.5 5.625   21.5H7.875C8.48438 21.5 9 21.0312 9 20.375V18.5H23.625C23.8125 18.5 24 18.3594 24 18.125V17.375C24 17.1875 23.8125 17 23.625 17ZM7.5 20H6V15.5H7.5V20ZM23.625 3.5H12V1.625C12 1.01562 11.4844 0.5 10.875 0.5H8.625C7.96875 0.5 7.5 1.01562 7.5 1.625V3.5H0.375C0.140625 3.5 0 3.6875 0 3.875V4.625C0 4.85938 0.140625 5 0.375 5H7.5V6.875C7.5 7.53125 7.96875 8 8.625 8H10.875C11.4844 8 12 7.53125 12 6.875V5H23.625C23.8125 5 24 4.85938 24 4.625V3.875C24 3.6875 23.8125 3.5 23.625 3.5ZM10.5 6.5H9V2H10.5V6.5ZM23.625 10.25H19.5V8.375C19.5 7.76562 18.9844 7.25 18.375 7.25H16.125C15.4688 7.25 15 7.76562 15 8.375V10.25H0.375C0.140625 10.25 0 10.4375 0 10.625V11.375C0 11.6094 0.140625 11.75 0.375 11.75H15V13.625C15 14.2812 15.4688 14.75 16.125 14.75H18.375C18.9844 14.75 19.5 14.2812 19.5 13.625V11.75H23.625C23.8125 11.75 24 11.6094 24 11.375V10.625C24 10.4375 23.8125 10.25 23.625 10.25ZM18 13.25H16.5V8.75H18V13.25Z\\" fill=\\"currentColor\\" />
</svg>
"
      />
    </View>
  </View>
</View>
`;
